### PR TITLE
ldns: conan v2 support

### PR DESCRIPTION
**Pull request by fcelda**
Modernize ldns for Conan 2.0.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/ldns/all/conanfile.py [hunk 1]
```diff
@@ -23,53 +29,58 @@ class LdnsConan(ConanFile):
...
+            # OpenSSL adds -lpthread to link with POSIX thread library. Instead, it should use -pthread compiler flag which additionally
+            # sets required macros at compile time.
+            tc.environment.remove("LIBS", "-lpthread")
+            tc.environment.append("CFLAGS", "-pthread")
+        except ValueError:
+            pass
+        tc.generate()
+
+        tc = AutotoolsToolchain(self)
         def yes_no(v): return "yes" if v else "no"
```
Comment (prince-chrismc) [inline recipes/ldns/all/conanfile.py:80]: Linter says its unused

```suggestion
```

File: recipes/ldns/all/test_v1_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.1)
```
Comment (prince-chrismc) [inline recipes/ldns/all/test_v1_package/CMakeLists.txt:1]: You should be able to drop the duplciated `example.c` if you redirect to the main test package https://github.com/conan-io/conan-center-index/blob/master/docs/package_templates/cmake_package/all/test_v1_package/CMakeLists.txt :)

File: recipes/ldns/all/conanfile.py (hunk 1 above)
Comment (fcelda) [reply recipes/ldns/all/conanfile.py]: I somehow missed it. It's fixed. Thanks.

File: recipes/ldns/all/test_v1_package/CMakeLists.txt (hunk 2 above)
Comment (fcelda) [reply recipes/ldns/all/test_v1_package/CMakeLists.txt:1]: Thanks for the suggestion. Done.

**Review by SSE4 — APPROVED** [review]

File: recipes/ldns/all/conanfile.py [hunk 3]
```diff
@@ -23,53 +29,57 @@ class LdnsConan(ConanFile):
...
+        tc = AutotoolsDeps(self)
+        try:
+            # This fixes an issue of linking against ldns in combination of openssl:shared=False, ldns:shared=True, and an older GCC:
+            # > hidden symbol `pthread_atfork' in /usr/lib/x86_64-linux-gnu/libpthread_nonshared.a(pthread_atfork.oS) is referenced by DSO
+            # OpenSSL adds -lpthread to link with POSIX thread library. Instead, it should use -pthread compiler flag which additionally
+            # sets required macros at compile time.
+            tc.environment.remove("LIBS", "-lpthread")
+            tc.environment.append("CFLAGS", "-pthread")
+        except ValueError:
+            pass
```
Comment (SSE4) [inline recipes/ldns/all/conanfile.py:76]: when does it raise and why?

Comment (fcelda) [reply recipes/ldns/all/conanfile.py:76]: It raises when `-lpthread` is not in `LIBS` and it happens in some environments with static build of OpenSSL.

**Pull request merged**